### PR TITLE
GeoServer interceptor - REST configuration and keycloak

### PR DESCRIPTION
**Pull request by ahennr**
* move entities to `shogun` / `interceptor` schema, rename flyway scripts
* keycloak config for geoserver interceptor
* Introduce REST endpoints to create / delete interceptor rules
* In order to change interceptor rules, an user must have the keycloak role `interceptor-admin`
  * otherwise: access to REST API is forbidden

Please review @terrestris/devs

### CODE REVIEW

File: shogun-boot/src/main/resources/db/migration/V0.1.0__Baseline.sql [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-CREATE SEQUENCE hibernate_sequence
+CREATE SEQUENCE IF NOT EXISTS hibernate_sequence
     INCREMENT 1
     START 1
     MINVALUE 1
     MAXVALUE 9223372036854775807
     CACHE 1;
 
-CREATE TABLE applications (
+CREATE TABLE shogun.applications (
```
Comment (dnlkoch) [inline shogun-boot/src/main/resources/db/migration/V0.1.0__Baseline.sql]: Migrations must not use a fixed schema, it should be configured by the appropriate jdbc/flyway config.

File: shogun-gs-interceptor/src/main/java/de/terrestris/shogun/interceptor/model/InterceptorRule.java [hunk 2]
```diff
@@ -6,12 +6,10 @@
...
 import lombok.*;
 
-import javax.persistence.Column;
-import javax.persistence.Entity;
-import javax.persistence.EnumType;
-import javax.persistence.Enumerated;
+import javax.persistence.*;
 
 @Entity
+@Table(schema = "interceptor")
```
Comment (dnlkoch) [inline shogun-gs-interceptor/src/main/java/de/terrestris/shogun/interceptor/model/InterceptorRule.java]: This should be configured in the jdbc config as well.

File: shogun-gs-interceptor/shogun-gs-interceptor-lib/src/main/java/de/terrestris/shogun/interceptor/enumeration/OgcEnum.java [hunk 3]
```diff
@@ -23,81 +23,75 @@
...
         w3dsOps.add(OperationType.GET_CAPABILITIES);
         w3dsOps.add(OperationType.GET_SCENE);
         w3dsOps.add(OperationType.GET_FEATURE_INFO);
         w3dsOps.add(OperationType.GET_LAYER_INFO);
         w3dsOps.add(OperationType.GET_TILE);
 
+        Set<OperationType> wmtsOps = new HashSet<>();
+        w3dsOps.add(OperationType.GET_CAPABILITIES);
+        w3dsOps.add(OperationType.GET_TILE);
+        w3dsOps.add(OperationType.GET_FEATURE_INFO);
```
Comment (dnlkoch) [inline shogun-gs-interceptor/shogun-gs-interceptor-lib/src/main/java/de/terrestris/shogun/interceptor/enumeration/OgcEnum.java]:
```suggestion
        wmtsOps.add(OperationType.GET_CAPABILITIES);
        wmtsOps.add(OperationType.GET_TILE);
        wmtsOps.add(OperationType.GET_FEATURE_INFO);
```

File: shogun-gs-interceptor/shogun-gs-interceptor-lib/pom.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+      </plugin>
+    </plugins>
+  </build>
+
+</project>
```
Comment (dnlkoch) [inline shogun-gs-interceptor/shogun-gs-interceptor-lib/pom.xml]: Missing newline.

**Comment by ahennr** [comment]
THX for reviews

**Pull request merged**